lib/resty/core/os.lua

```diff
@@ -0,0 +1,46 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local base = require "resty.core.base"
+base.allows_subsystem('http')
+
+
+local ffi = require "ffi"
+local os = require "os"
+local C = ffi.C
+local ffi_str = ffi.string
+local getfenv = getfenv
+local error = error
+
+
+ffi.cdef[[
+
+struct ngx_cycle_s;
+typedef struct ngx_cycle_s  ngx_cycle_t;
+
+char *ngx_http_lua_ffi_os_getenv(ngx_cycle_t *cycle, ngx_http_request_t *r,
+    const char *varname);
+]]
+
+
+local _M = { version = base.version }
+
+
+function os.getenv(varname)
+    local t = getfenv(0)
+
+    local cycle = t.__ngx_cycle
+    if not cycle then
+        error("no cycle found")
+    end
+
+    local r = t.__ngx_req
+
+    local value = C.ngx_http_lua_ffi_os_getenv(cycle, r, varname)
+    if value then
```

Comment:
The check here is incorrect.
In Lua, only `false` and `nil` are falsy value, so `cdata:NULL` is a truth value.
Need to check it with `if value ~= nil`.
BTW, better to add a test case here.

Reply:
@spacewander 
Okay, i have test this, in LuaJIT `cdata<char *>: NULL` is truth, we can use `if cdata == nil` to check if `cdata` is a NULL pointer, `ffi.string(nil)` and `ffi.string(cdata<char *>: NULL)` 
will result segmentation fault, thanks!